jupyterlab/labextensions.py

```diff
@@ -0,0 +1,1086 @@
+
+    Parameters
+    ----------
+    module : str
+        Importable Python module exposing the
+        magic-named `_jupyter_lab_extension_paths` function
+    user : bool [default: True]
+        Whether to enable in the user's lab_extensions directory.
+    sys_prefix : bool [default: False]
+        Whether to enable in the sys.prefix, i.e. environment
+    logger : Jupyter logger [optional]
+        Logger instance to use
+    """
+    return _set_lab_extension_state_python(False, module, user, sys_prefix,
+                                         logger=logger)
+
+
+def validate_lab_extension(name, logger=None):
+    """Validate a named lab_extension.
+
+    Looks across all of the lab_extension directories.
+
+    Returns a list of warnings.
+
+    Parameters
+    ----------
+    name : str
+        The name of the extension.
+    logger : Jupyter logger [optional]
+        Logger instance to use
+    """
+    for exts in _lab_extension_dirs():
+        full_dest = os.path.join(exts, name)
+        if os.path.exists(full_dest):
+            return validate_lab_extension_folder(full_dest)
+
+
+def validate_lab_extension_folder(name, full_dest, logger=None):
+    """Assess the health of an installed lab_extension
+
+    Returns a list of warnings.
+
+    Parameters
+    ----------
+    full_dest : str
+        The on-disk location of the installed lab_extension: this should end
+        with `lab_extensions/<name>`
+    logger : Jupyter logger [optional]
+        Logger instance to use
+    """
+    if logger:
+        logger.info("    - Validating...")
+
+    infos = []
+    warnings = []
+
+    hasFiles = True
+    hasEntry = False
+    data = get_lab_extension_manifest_data(full_dest)
+    for manifest in data.values():
+        if ('entry' in manifest and 'modules' in manifest):
+            if (manifest['entry'] in manifest['modules']):
+                hasEntry = True
+        files = manifest.get('files', [])
+        if not files:
+            hasFiles = False
+        for fname in files:
+            path = os.path.join(full_dest, fname)
+            if not os.path.exists(path):
+                hasFiles = False
+    entry_msg = u"     {} has {} entry point?"
+    name = os.path.basename(full_dest)
+    if hasEntry:
+        (entry_msg.format(GREEN_OK, name))
+    else:
+        warnings.append(entry_msg.format(RED_X, name))
+
+    files_msg = u"     {} has necessary files?"
+    if hasFiles:
+        infos.append(files_msg.format(GREEN_OK, name))
+    else:
+        warnings.append(files_msg.format(RED_X, name))
+
+    post_mortem = u"      {} {} {}"
+    if logger:
+        if warnings:
+            [logger.info(info) for info in infos]
+            [logger.warn(warning) for warning in warnings]
+        else:
+            logger.info(post_mortem.format(name, "", GREEN_OK))
+
+    return warnings
+
+
+#----------------------------------------------------------------------
```

Comment:
Are you not done with this file yet? Below here, there are lots of mentions of nb extensions - not sure if it's dead unnecessary code, or a search and replace s/NB/Lab/ is in order.